src/utils/types.ts

```diff
@@ -0,0 +1,46 @@
+export type Character = {
+  id: string;
+  name: string;
+  species: string;
+};
+
+export type CharactersData = {
+  characters: {
+    results: Character[];
+    info: {
+      next: string;
+    };
+  };
+};
+
+export type Episode = {
+  name: string;
+  episode: string;
+};
+
+export type CharacterDetails = {
+  id: string;
+  image: string;
+  name: string;
+  species: string;
+  status: string;
+  gender: string;
+  location: {
+    name: string;
+  };
+  origin: {
+    name: string;
+  };
+  episode: {
+    name: string;
+    episode: string;
+  }[];
+};
+
+export type CharacterListProps = {
```

Comment:
This kind of types must be defined inside of it's own component is not something you are going to reuse in other ones